xds/src/test/java/io/grpc/xds/sds/trust/SdsX509TrustManagerTest.java

```diff
@@ -221,14 +244,132 @@ public void oneIpAddressInPeerCertsMismatch() throws CertificateException, IOExc
             .addVerifySubjectAltName("x.foo.com")
             .addVerifySubjectAltName("192.168.2.3")
             .build();
-    SdsX509TrustManager trustManager = new SdsX509TrustManager(certContext, mockDelegate);
+    trustManager = new SdsX509TrustManager(certContext, mockDelegate);
     X509Certificate[] certs =
         CertificateUtils.toX509Certificates(TestUtils.loadCert(SERVER_1_PEM_FILE));
     try {
       trustManager.verifySubjectAltNameInChain(certs);
-      Assert.fail("no exception thrown");
+      fail("no exception thrown");
     } catch (CertificateException expected) {
       assertThat(expected).hasMessageThat().isEqualTo("Peer certificate SAN check failed");
     }
   }
+
+  @Test
+  public void checkServerTrustedSslEngine()
+      throws CertificateException, IOException, CertStoreException {
+    TestSslEngine sslEngine = buildTrustManagerAndGetSslEngine();
+    X509Certificate[] serverCerts =
+        CertificateUtils.toX509Certificates(TestUtils.loadCert(SERVER_1_PEM_FILE));
+    trustManager.checkServerTrusted(serverCerts, "ECDHE_ECDSA", sslEngine);
+    verify(sslEngine, times(1)).getHandshakeSession();
```

Comment:
fyi, times(1) can be omitted unless very good reason to emphasize times(1). i am just pointing out because all verify has it.